### ISSUE TITLE
Privacy mode doesn't hide favorites from the api

### ISSUE THREAD

**Issue by nonamethanks** [r888888888/danbooru#3633]
Example:
https://danbooru.donmai.us/posts/3083933 - 13 users shown 
https://danbooru.donmai.us/posts/3083933.xml - 15 users shown

There's probably more places where this information is still leaking so it needs to be looked into more.

**Comment by evazion** [comment]
Fixing this would require loading every user to see who has privacy mode enabled. This could slow down the API. For single posts it's not so bad, but for a search like `/posts.json?tags=hews_hack+limit:200`, which has 3000+ unique favoriters and 16000+ total favorites, it could be an issue.

**Comment by BrokenEagle** [comment]
Why not just strip out the ``fav_string`` on index calls if you're not a Moderator+?  That information isn't being included in the index calls for the HTML anyways.  I realize that this might make some things more difficult for certain uses of the API, but it'd avoid the slowdown issue for large index API calls.  Also, I can't really think of any legitimate uses for mass checking the post favorites using the API like this except for perhaps stalking another user.

**Comment by evazion** [comment]
That would work. It will break compatibility if anyone does need the fav_string, but I don't think that can be helped.

**Comment by BrokenEagle** [comment]
If it's needed, a blank string could be shown instead.

**Comment by Moebius-Strip** [comment]
Unless I'm mistaken, the favorites meta tag also exposes a full list of favorites. Should it be hidden as well or even just removed if it's unnecessary? If so, anything that makes use of it would need updating as well. All I noticed was a portion of the favorites JavaScript using it. I created [a little branch](https://github.com/r888888888/danbooru/compare/favorites-meta-privacy?expand=1) just to point out what I'm talking about.

With the removal of the favorites string, I'd also like to suggest including the "is favorited" information ( post.favorited_by?(CurrentUser.user.id) ) in the API data since it can be useful to know which posts the current user has favorited if you want to potentially do things like excluding favorites from the blacklist or distinguishing favorites from other posts in a listing.
